pybop/models/spm.py

```diff
@@ -1,21 +1,12 @@
-import pybop
 import pybamm
 from .base_model import BaseModel
 
-class BaseSPM():
+class BaseSPM(pybamm.models.full_battery_models.lithium_ion.BasicSPM):
```

Comment:
There are two main methods to establish relationships between classes in OO, inheritance (which you are using), and composition. See here for more details: https://train.oxrse.uk/material/software_architecture_and_design/object_orientated/inheritance_and_composition_cpp

Composition tends to be more flexible as you don't lock yourself into an inflexible type hierarchy. This is especially disadvantageous if you are inheriting from a class you do not control.

So I would use composition here. Something like this:

```python
class SPM():
  def __init__(self):
    self.pybamm_model = pybamm.lithium_ion.SPM()
    self.name = "SPM"

Reply:
Excellent, thanks for the feedback @martinjrobins.